infoblox/resource_infoblox_a_record.go

```diff
@@ -254,6 +254,31 @@ func resourceARecordUpdate(d *schema.ResourceData, m interface{}) error {
 		return fmt.Errorf("error updating A-record: either 'ip_addr' or 'cidr' value must not be empty")
 	}
 
+	oldCidr, _ := d.GetChange("cidr")
+
+	if oldCidr == "" {
+		if d.HasChange("cidr") && !d.HasChange("ip_addr") {
```

Comment:
What will happen is `cidr` becomes non-empty but `ip_addr` was not empty and stays unchanged?